python/ray/experimental/serve/server.py

```diff
@@ -101,6 +112,59 @@ def __init__(self):
 
         return b"".join(body_buffer)
 
+    async def handle_profiling_requests(self, current_path, scope, receive,
+                                        send):
+        profile_string, service_name = os.path.split(current_path)
+        if (profile_string != SERVE_PROFILE_PATH
+                or service_name not in self.policy_table_cache):
+            error_message = ("Path {} not found. Please enter"
+                             " the correct path for profiling the"
+                             " service".format(current_path))
+            await JSONResponse(
+                {
+                    "error": error_message
+                }, status_code=404)(scope, receive, send)
+            return
+
+        # load the kwargs creator
+        kwargs_creator = pickle.loads(self.policy_table_cache[service_name])
```

Comment:
this will slow down the web server by a lot. pickle loads for each query can increase per-query latency by quite a lot and bring down the throughput. consider cache kwargs_creator?

Reply:
The time being noted is after the kwargs are loaded. Anyhow I will consider a caching mechanism. Good point